audio/out/ao_wasapi_utils.c

```diff
@@ -576,8 +576,8 @@ static void init_session_display(struct wasapi_state *state, const char *name) {
                                          (void **)&state->pSessionControl);
     EXIT_ON_ERROR(hr);
 
-    wchar_t path[MAX_PATH] = {0};
-    GetModuleFileNameW(NULL, path, MAX_PATH);
+    wchar_t path[MP_PATH_MAX] = {0};
```

Comment:
General remark to this PR. I don't think it is reasonable to allocate 32k of wchars on stack.

Reply:
maybe. I was thinking I could just say it's windows fault for having stupidly high limits.
or want me to change it all to talloc?

Reply:
`GetModuleFileNameW` is annoying cookie, because it doesn't follow normal MS pattern.

But in other cases, we can do like `GetCurrentDirectory(0, NULL)` to determine buffer size and talloc this. And this pattern should be applied to all long path aware functions.

For `GetModuleFileNameW` I would suggest keeping `MAX_PATH` on stack as it is now. And check if we got `ERROR_INSUFFICIENT_BUFFER` then talloc 32k as a fallback, which should almost never happen in practice.

Reply:
> But in other cases, we can do like `GetCurrentDirectory(0, NULL)` to determine buffer size and talloc this. And this pattern should be applied to all long path aware functions.

I'll check and do that.

> For GetModuleFileNameW I would suggest keeping MAX_PATH on stack as it is now. And check if we got ERROR_INSUFFICIENT_BUFFER then talloc 32k as a fallback, which should almost never happen in practice.

Unnecessary complexity IMO, none of this code is in a hot path and unconditionally allocating 64k bytes doesn't hurt.